jf_agent/main.py

```diff
@@ -136,21 +137,19 @@ def main():
             sys_diag_collector.join()
 
         finally:
-            write_file(config.outdir, 'status', config.compress_output_files, status_json)
             diagnostics.close_file()
 
     if config.run_mode_includes_send:
         send_data(
-            config.outdir,
-            jellyfish_endpoint_info.s3_uri_prefix,
-            jellyfish_endpoint_info.aws_access_key_id,
-            jellyfish_endpoint_info.aws_secret_access_key,
+            config,
+            creds
         )
     else:
         print(f'\nSkipping send_data because run_mode is "{config.run_mode}"')
         print(f'You can now inspect the downloaded data in {config.outdir}')
         print(f'To send this data to Jellyfish, use "-m send_only -od {config.outdir}"')
 
+    write_file(config.outdir, 'status', config.compress_output_files, status_json)
```

Comment:
Don't we want to send this to the backend?  Should that happen befire `send_data`?